clients/backend-client-elixir/lib/adf_sender_connector/channel.ex

```diff
@@ -0,0 +1,49 @@
+defmodule AdfSenderConnector.Channel do
+  @moduledoc """
+  Async Dataflow Rest client for /ext/channel/create endpoint
+  """
+
+  use AdfSenderConnector.Spec
+
+  @doc """
+  Request Channel Sender to register a channel for the application and user indicated.
+  """
+  @spec create_channel(pid(), application_ref(), user_ref()) :: {:ok, any()} | {:error, any()}
+  def create_channel(pid, application_ref, user_ref) do
+    GenServer.call(pid, {:create_channel, application_ref, user_ref})
+  end
+
+  ##########################
+  # Server Implementation  #
+  ##########################
+
+  @doc false
+  @impl true
+  def handle_call({:create_channel, application_ref, user_ref}, _ctx, state) do
+    response = build_new_channel_request(application_ref, user_ref)
+    |> request_channel_creation(state)
+    |> decode_response
+
+    {:reply, response, state}
+  end
+
+  defp build_new_channel_request(application_ref, user_ref) do
+    Jason.encode!(%{
+      application_ref: application_ref,
+      user_ref: user_ref
+    })
+  end
+
+  defp request_channel_creation(request, state) do
+    HTTPoison.post(
+      Keyword.fetch!(state, :sender_url) <> "/ext/channel/create",
+      request,
+      [{"Content-Type", "application/json"}],
+      hackney: [:insecure, pool: :default],
```

Comment:
Should be insecure?

Reply:
You're right, and definitively not by default. I made possible to provide hackney/httpoison options to the process and then apply them at the request.

https://github.com/bancolombia/async-dataflow/blob/dc4c52b1db2082fff8aeccee3a688959de1eca77/clients/backend-client-elixir/lib/adf_sender_connector/channel.ex#L42